### PR TITLE
Add basic RUM conversion tracking

### PR DESCRIPTION
**Pull request by herzog31**
* Add basic RUM conversion tracking as used in Maidenform

Test URLs:
- Before: https://main--aem-boilerplate-commerce--hlxsites.hlx.live/
- After: https://conversion--aem-boilerplate-commerce--hlxsites.hlx.live/

### CODE REVIEW

**Review by dicagno — APPROVED** [review]

**Review by meryllblanchet — APPROVED** [review]
LGTM, thanks @herzog31 

@chicharr @ekremney please give a last pair of eyes.

**Comment by ekremney** [comment]
@herzog31 LGTM. I have two non-blocking comments:

1. `source` and `target` values seem to be the same for all the conversion events in the PR: for example, `convert('createAccount', 'createAccount')`. `source` is the conversion event, and the `target` is the conversion value. If there is no meaningful "conversion value", then we may as well leave the `target` field empty

2. if possible (and available), it would be nice to add the `element` field in the conversion event to see which element (ie button) generated the conversion

File: scripts/delayed.js [hunk 1]
```diff
@@ -37,3 +37,30 @@ if (getConsent('commerce-collection')) {
...
 }
+
+// Conversion tracking
+function convert(event, extraData = {}) {
+  sampleRUM('convert', { source: event, ...extraData });
+}
+
+window.adobeDataLayer.push((dl) => {
+  dl.addEventListener('add-to-cart', ({ eventInfo }) => {
+    convert('addToCart', { skus: eventInfo?.shoppingCartContext?.items?.map((item) => item.product.sku), element: 'cart' });
```
Comment (chicharr) [inline scripts/delayed.js:48]: `skus` and `element` will be discarded. 
We only send to backend known properties. See https://github.com/adobe/helix-rum-js/blob/0bc640edd24944c77303b7103336306850012f8e/src/index.js#L72 

you can squeeze them in the `target` field

The checkpoints is also filtered on the [collector side](https://github.com/adobe/helix-rum-collector/blob/bb1189ca43682b22e863e804257cc85880bdf8fe/src/utils.mjs#L16C17-L16C34), you would need to add these checkpoints to the list

**Review by chicharr — CHANGES_REQUESTED** [review]

File: scripts/delayed.js (hunk 1 above)
Comment (herzog31) [reply scripts/delayed.js:48]: Could you give an example of how the addToCart RUM event should look like to meet your requirements?

Comment (chicharr) [reply scripts/delayed.js:48]: Something like this
```
convert('addToCart', { 
  target: eventInfo?.shoppingCartContext?.items?.map((item) => item.product.sku).join() 
});
```

Comment (chicharr) [reply scripts/delayed.js:48]: but you still need to whitelist the `addToCart` checkpoint in the collector

Comment (herzog31) [reply scripts/delayed.js:48]: @chicharr be aware that the `convert` function results into this:

```javascript
sampleRUM('convert', { source: 'addToCart', target: eventInfo?.shoppingCartContext?.items?.map((item) => item.product.sku).join()  });
```

Does that work for you or would you rather have `addToCart` as dedicated checkpoint like this?

```javascript
sampleRUM('addToCart', { target: eventInfo?.shoppingCartContext?.items?.map((item) => item.product.sku).join() });
```

**Pull request closed without merging**